front/lib/api/wrappers.ts

```diff
@@ -361,3 +362,67 @@ export function withAuth0TokenAuthentication<T>(
     }
   );
 }
+
+// This is a type that represents the resources that can be extracted from an API route
+type RouteResourceName = "space";
+
+type RouteResourceMap = {
+  space: SpaceResource;
+};
+
+/*
+ *  API routes containing resource strings that require some handling logic can
+ *  use this wrapper to extract the resource, make the checks, apply the logic
+ *  and then call the handler with the resource.
+ *
+ * e.g. for /w/[wId]/spaces/[spaceId]/... => check the space exists, that it's
+ *  not a conversation space, etc. and provide the space resource to the handler.
+ */
+export function withInternalAPIRouteResource<T, U extends RouteResourceName>(
```

Comment:
I like the idea of introducing a wrapper to fetch a resource from the route. However, I'd try to keep it separated from the authentification to avoid having one helper that does both.

For example, we could introduce a `withResourceFetching` wrapper that can be used afterward.

It could looks like this:

```
const handler = withResourceFetching(async (
  req: NextApiRequest,  
  res: NextApiResponse<WithAPIErrorResponse<GetDatasetResponseBody  | PostDatasetResponseBody>>,
  auth: Authenticator,
  space: SpaceResource
) => {
  const { aId, spaceId } = req.query;
  if (typeof spaceId !== "string" || typeof aId !== "string") {
    return apiError(req, res, {
      status_code: 400,
      api_error: {
        type: "invalid_request_error",
        message: "Invalid path parameters.",
      },
    });
  }

  // Rest of the code....
}, SpaceResource);

export default withSessionAuthenticationForWorkspace(handler);
```

Any reason to pass a literal string over a resource itself?

Reply:
I understand the point, and would agree, I tried to do that. But the issue is, withResourceFetching is actually subtly tied to the auth mechanism . If we want a very general wrapper, then the args of the inner handler are auth & session for an internal route, while they're auth & keyAuth for public route. Since the tying is there, I thought it was better to have it explicitely

literal string VS resource => agreed, I can use a resource

Reply:
I see your point on the type but IIRC we already have this type available so we can definitely use it in the `withResourceFetching` helper, no?

Reply:
Discussed IRL, updated the PR